### PR TITLE
Add Resource List Tab

### PR DESCRIPTION
**Pull request by gagelarsen**
**Primary changes in this Pull Request:**

- Add Resource List Tab

**Please review the checklist before submitting the Pull Request:**

- [x] Pull request has a meaning full name (not just the commit message from of your last commit)
- [x] Code has been linted using flake8
- [x] All methods have accurate Google-Style Docstrings
- [x] 100% test coverage for new content
- [x] Tests for each common use case (please don't write one test that covers all use cases)
- [ ] Bonus: Use TypeHints

**Explain deviations from original design if applicable:**

- N/A

### CODE REVIEW

File: tethysext/atcore/controllers/resources/tabs/resource_list_tab.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    Methods:
+        get_resources (required): Get a list of resources to be associated with this resource.
+    """  # noqa: E501
+    template_name = 'atcore/resources/tabs/resource_list_tab.html'
+    base_template = None
+    post_load_callback = 'resource_list_tab_loaded'
+
+    js_requirements = ResourceTab.js_requirements + [
+        'atcore/resources/resource_list_tab.js',
+        # TODO: Remove if we don't use this.
```
Comment (swainn) [inline tethysext/atcore/controllers/resources/tabs/resource_list_tab.py:31]: Remove if this is no longer a todo.

File: tethysext/atcore/controllers/resources/tabs/resource_list_tab.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    ]
+    css_requirements = ResourceTab.css_requirements + [
+        'atcore/resources/resource_list_tab.css'
+        # TODO: Remove if we don't use this.
```
Comment (swainn) [inline tethysext/atcore/controllers/resources/tabs/resource_list_tab.py:35]: Remove todo comment

File: tethysext/atcore/controllers/resources/tabs/resource_list_tab.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    ]
+
+    def get_resources(self, request, resource, session, *args, **kwargs):
+        """
+        Get a list of resources
+
+        Returns:
+            A list of FileCollection clients.
```
Comment (swainn) [inline tethysext/atcore/controllers/resources/tabs/resource_list_tab.py:43]: "A list of Resources.￼"

File: tethysext/atcore/controllers/resources/tabs/resource_list_tab.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        for resource in resources:
+            resource_card = resource.__dict__
+            resource_card['debugging'] = resource.attributes
+            resource_card['debugging']['id'] = str(resource.id)
+
+            resource_card['action'] = 'launch'
+            resource_card['action_title'] = 'View Resource'
+            resource_card['action_href'] = reverse(f'{self._app.namespace}:{resource.SLUG}_resource_details',
+                                                   args=[resource.id])
+            print(resource_card['action_href'])
```
Comment (swainn) [inline tethysext/atcore/controllers/resources/tabs/resource_list_tab.py:58]: Remove print or convert to log.

File: tethysext/atcore/controllers/resources/tabs/resource_list_tab.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        """
+        Get a list of resources
+
+        Returns:
+            A list of FileCollection clients.
+        """
+        return []
+
+    def _build_resource_cards(self, resources):
+        resource_cards = []
```
Comment (swainn) [inline tethysext/atcore/controllers/resources/tabs/resource_list_tab.py:48]: Doc string?

File: tethysext/atcore/tests/integrated_tests/controllers/resources/tabs/resource_list_tab_tests.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+def setUpModule():
+    setup_module_for_sqlalchemy_tests()
+
+
+def tearDownModule():
+    tear_down_module_for_sqlalchemy_tests()
+
+
+class ResourceListTabWithDatabase(ResourceListTab):
```
Comment (swainn) [inline tethysext/atcore/tests/integrated_tests/controllers/resources/tabs/resource_list_tab_tests.py:28]: I don't think this is used is it?

File: tethysext/atcore/tests/integrated_tests/controllers/resources/tabs/resource_list_tab_tests.py [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    def test_get_no_resources(self):
+        """Test get_context()"""
+        instance = ResourceListTab()
+        request = self.request_factory.get('/foo/12345/bar/summary/')
+        request.user = self.get_user()
+        instance._app = mock.MagicMock()
+        context = instance.get_context(request, self.session, self.resource, {})
+
+        self.assertIn('resources', context)
+        self.assertEqual(len(context['resources']), 0)
```
Comment (swainn) [inline tethysext/atcore/tests/integrated_tests/controllers/resources/tabs/resource_list_tab_tests.py:76]: Tests for other methods? Need tests for get_resources and _build_resource_cards.

File: tethysext/atcore/templates/atcore/resources/tabs/resource_list_tab.html [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+                  <td class="btn-td">
+                    <div class="btn-toolbar pull-right">
+                        <a class="btn btn-s btn-primary btn-flat btn-action btn-resource-launch" href="{{ resource.action_href }}" data-toggle="tooltip" data-placement="top" title="{{ resource.action_title }}"><span class="glyphicon glyphicon-chevron-right"></span></a>
+                    </div>
+                  </td>
+                </tr>
+              {% endfor %}
+            </tbody>
+          </table>
+        </div>
```
Comment (swainn) [inline tethysext/atcore/templates/atcore/resources/tabs/resource_list_tab.html:65]: Remove extra </div> and clean up indenting.

File: tethysext/atcore/templates/atcore/app_users/manage_resources.html [hunk 9]
```diff
@@ -9,7 +9,10 @@
       <a class="btn btn-default manage-action-btn btn-new" href="{% url base_resource_href|add:'_new_resource' %}" data-toggle="tooltip" data-placement="bottom" title="New"><span class="glyphicon glyphicon-plus"></span></a>
     </div>
     {% endif %}
-    <h1>{{ page_title }}</h1>
+
+    {% if page_title %}
```
Comment (swainn) [inline tethysext/atcore/templates/atcore/app_users/manage_resources.html:13]: Why this change?

File: tethysext/atcore/controllers/resources/tabs/resource_list_tab.py [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+        for resource in resources:
+            resource_card = resource.__dict__
+            resource_card['debugging'] = resource.attributes
+            resource_card['debugging']['id'] = str(resource.id)
```
Comment (swainn) [inline tethysext/atcore/controllers/resources/tabs/resource_list_tab.py:52]: I don't think this debugging information is used is it? Remove if not.

**Review by swainn — CHANGES_REQUESTED** [review]
Is resource_list_tab_base.html needed?

File: tethysext/atcore/controllers/resources/tabs/resource_list_tab.py (hunk 10 above)
Comment (gagelarsen) [reply tethysext/atcore/controllers/resources/tabs/resource_list_tab.py:52]: I think you are right. removed.

File: tethysext/atcore/templates/atcore/app_users/manage_resources.html (hunk 9 above)
Comment (gagelarsen) [reply tethysext/atcore/templates/atcore/app_users/manage_resources.html:13]: Sorry, holdover from a way i was thinking of implementing this.

File: tethysext/atcore/templates/atcore/resources/tabs/resource_list_tab.html (hunk 8 above)
Comment (gagelarsen) [reply tethysext/atcore/templates/atcore/resources/tabs/resource_list_tab.html:65]: don't know how that slipped through...

File: tethysext/atcore/tests/integrated_tests/controllers/resources/tabs/resource_list_tab_tests.py (hunk 6 above)
Comment (gagelarsen) [reply tethysext/atcore/tests/integrated_tests/controllers/resources/tabs/resource_list_tab_tests.py:28]: good catch

File: tethysext/atcore/controllers/resources/tabs/resource_list_tab.py (hunk 5 above)
Comment (gagelarsen) [reply tethysext/atcore/controllers/resources/tabs/resource_list_tab.py:48]: Done

File: tethysext/atcore/controllers/resources/tabs/resource_list_tab.py (hunk 4 above)
Comment (gagelarsen) [reply tethysext/atcore/controllers/resources/tabs/resource_list_tab.py:58]: sorry, thought i removed that.

File: tethysext/atcore/controllers/resources/tabs/resource_list_tab.py (hunk 3 above)
Comment (gagelarsen) [reply tethysext/atcore/controllers/resources/tabs/resource_list_tab.py:43]: whoops

File: tethysext/atcore/controllers/resources/tabs/resource_list_tab.py (hunk 2 above)
Comment (gagelarsen) [reply tethysext/atcore/controllers/resources/tabs/resource_list_tab.py:35]: removed

File: tethysext/atcore/controllers/resources/tabs/resource_list_tab.py (hunk 1 above)
Comment (gagelarsen) [reply tethysext/atcore/controllers/resources/tabs/resource_list_tab.py:31]: removed

File: tethysext/atcore/tests/integrated_tests/controllers/resources/tabs/resource_list_tab_tests.py (hunk 7 above)
Comment (gagelarsen) [reply tethysext/atcore/tests/integrated_tests/controllers/resources/tabs/resource_list_tab_tests.py:70]: added

File: tethysext/atcore/controllers/resources/tabs/resource_list_tab.py [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+********************************************************************************
+"""
+from django.shortcuts import reverse
+
+from .resource_tab import ResourceTab
+
+
+class ResourceListTab(ResourceTab):
+    """
+    # A tab for the TabbedResourceDetails view that lists resources related to this resource.
```
Comment (swainn) [inline tethysext/atcore/controllers/resources/tabs/resource_list_tab.py:16]: Comment in a comment? Remove `# `

File: tethysext/atcore/tests/integrated_tests/controllers/resources/tabs/resource_list_tab_tests.py [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+        with mock.patch.object(ResourceListTab, 'get_resources') as mock_get_resources:
+            model = SpatialResource()
+            mock_get_resources.return_value = [model]
+            instance._app = mock.MagicMock()
+            context = instance.get_context(request, self.session, self.resource, {})
+
+        self.assertIn('resources', context)
+        self.assertEqual(len(context['resources']), 1)
+
+    def test_get_no_resources(self):
```
Comment (swainn) [inline tethysext/atcore/tests/integrated_tests/controllers/resources/tabs/resource_list_tab_tests.py:61]: `test_get_context_no_resources`?

**Review by swainn — CHANGES_REQUESTED** [review]

**Review by swainn — APPROVED** [review]
